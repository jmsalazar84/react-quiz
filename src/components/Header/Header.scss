.header {
  background-color: #512da8;
  color: #fff;
  display: flex;
  width: 100%;
}

.title {
  flex: 0 1 30%;
  font-weight: bold;
  padding: 1em 2em;
  text-align-last: left;
  white-space: nowrap;
}

.links {
  flex: 0 1 70%;
  padding: 1em 2em;
  text-align-last: right;
  white-space: nowrap;

  a {
    color: #fff;
  }

  a:hover {
    color: #fff;
  }
}
