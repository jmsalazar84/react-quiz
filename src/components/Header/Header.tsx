import React from 'react';
import { NavLink } from 'react-router-dom';
import style from './Header.scss';

export interface HeaderProps {
  title: string;
  links: { link: string; text: string }[];
}

export const Header: React.FC<HeaderProps> = ({ title, links }: HeaderProps): JSX.Element => {
  const items = links.map((item, index) => (
    <a href={item.link} key={index}>
      {item.text}
    </a>
  ));
  return (
    <div className={style.header}>
      <div className={style.title}>{title}</div>
      <div className={style.links}>{items}</div>
    </div>
  );
};
