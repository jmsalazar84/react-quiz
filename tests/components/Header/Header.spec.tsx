import React from 'react';
import renderer from 'react-test-renderer';

import { HeaderProps, Header } from '@components/Header/Header';

describe('Header component', () => {
  test('should render component properly', () => {
    // given
    const headerProps = {
      title: 'Tile',
      links: [{ text: 'Link', link: '/' }],
    } as HeaderProps;

    // when
    const componentRenderer = renderer.create(<Header {...headerProps} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
