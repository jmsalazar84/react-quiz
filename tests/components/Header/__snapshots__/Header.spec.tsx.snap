// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render component properly 1`] = `
<div
  className="header"
>
  <div
    className="title"
  >
    Tile
  </div>
  <div
    className="links"
  >
    <a
      href="/"
    >
      Link
    </a>
  </div>
</div>
`;
